src/orion/benchmark/task/task.py

```diff
@@ -0,0 +1,298 @@
+        self.seed = seed
+        fixed_dims = fixed_dims or {}
+        # Fixed dimensions in the hparam space.
+        self.fixed_values: Dict[str, Any] = {
+            key: value
+            for key, value in fixed_dims.items()
+            if key in self.hparams.get_orion_space_dict()
+        }
+        logger.debug(
+            f"Creating a Task of type {type(self).__name__}, with task id "
+            f"{task_id} and with fixed values {self.fixed_values}"
+        )
+
+    @abstractmethod
+    def __call__(
+        self, hp: Union[HParams, Dict, np.ndarray] = None, **kwargs
+    ) -> np.ndarray:
+        """Evaluates the given samples and returns the performance.
+
+        Args:
+            hp (Union[HyperParameters, Dict, np.ndarray], optional):
+                Either a Hyperparameter dataclass, a dictionary, or a numpy
+                array containing the values for each dimension. Defaults to
+                None.
+
+        Returns:
+            np.ndarray: The performances of the hyperparameter on the sampled
+            task.
+        """
+
+        # return super().__call__(**kwargs)
+
+    def call(self, *args, **kwargs):
+        return self(*args, **kwargs)
+
+    @property
+    def max_trials(self) -> int:
+        """Return the max number of trials to run with this objective."""
+        return self._max_trials
+
+    def get_search_space(self) -> Dict[str, str]:
+        """Return the search space for the task objective function"""
+        return self.space
+
+    @property
+    def configuration(self):
+        """Return the configuration of the task."""
+        return {
+            self.__class__.__qualname__: {
+                "task_id": self.task_id,
+                "seed": self.seed,
+                "fixed_values": self.fixed_values,
+                "max_trials": self.max_trials,
+            }
+        }
+
+    @property
+    def full_space(self) -> Dict[str, str]:
+        """ Auto-generated orion-style 'space' dict for `self.hparams`.
+
+        This is generated using the `self.hparams` dataclass type.
+
+        Returns:
+            Dict[str, str]: Dict mapping from dimension name to the 'space'.
+        """
+        return self.hparams.get_orion_space_dict()
+
+    @classmethod
+    def get_domain(cls) -> Domain:
+        bounds = cls.hparams.get_bounds()
+        return Domain(
+            lower=np.asfarray([bound.domain[0] for bound in bounds]),
+            upper=np.asfarray([bound.domain[1] for bound in bounds]),
+        )
+
+    @property
+    def space(self) -> Dict[str, str]:
+        """ Auto-generated orion-style 'space' strings for `self.hparams`.
+
+        Entries from `self.fixed_values` are removed from the returned dictionary.
+        """
+        space = self.full_space.copy()
+        for k in self.fixed_values:
+            space.pop(k)
+        return space
+
+    @property
+    def full_domain(self) -> List[Dict[str, Any]]:
```

Comment:
Why do we need GPopt like domain definitions instead of only Oríon space definitions?

Reply:
Removed those.